sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -265,80 +211,85 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
             return reject(err);
           }
         }
-        const timeTakenByInit = Date.now() - initStartTime;
-
-        log.sender(
-          "[%s] Sender '%s', credit: %d available: %d",
-          this._context.connectionId,
-          this.name,
-          this.link!.credit,
-          this.link!.session.outgoing.available()
-        );
 
-        if (!this.link!.sendable()) {
-          log.sender(
-            "[%s] Sender '%s', waiting for 1 second for sender to become sendable",
-            this._context.connectionId,
-            this.name
-          );
-
-          await delay(1000);
+        try {
+          const timeTakenByInit = Date.now() - initStartTime;
 
           log.sender(
-            "[%s] Sender '%s' after waiting for a second, credit: %d available: %d",
+            "[%s] Sender '%s', credit: %d available: %d",
             this._context.connectionId,
             this.name,
-            this.link!.credit,
-            this.link!.session.outgoing.available()
+            this.link?.credit,
```

Comment:
Do we expect this.link to be null/undefined at this point in any scenario? Or is this just to escape TS checks? If it is the latter what makes us choose ? over !

Reply:
If a close happens externally from us (say from detached) _after_ init() then it can close the link before we get here (was running into this in the tests).

Reply:
So, in such cases, we end up in the else block and throw a `SenderBusyError` with  `cannot send the message right now. Please try later` message. Wouldnt we want to throw some other kind of error?

Reply:
Filed as part of #10921